### DISCUSSION TITLE
Implementing custom validation for a type i64 but by consuming a &str

### DISCUSSION THREAD

**Discussion by srfsh** [jprochazk/garde#124] [General]
In my application I store IDs in the underlying form of i64. However, they are presented to the users in the surface form of string. Now, I won't be, obviously, getting any i64 forms from the outside, since the underlying form never leaves the system. The only way they can enter the system is through their surface form, which is string.

I want to be able to validate the format of the string, while still being able to use it as i64 using the newtype convention. This might not strictly be validation, but rather a combination of validation and parsing, but it is really useful for my application in terms of DX. I was wondering if there is a way to achieve this using Garde. 

Many thanks \o/

**Comment by jprochazk** [comment]
Garde does not handle parsing, validation is done on fully initialized types. It seems to me like you want to combine deserialization using `serde` and then calling `validate` on the deserialized type.
One possible source of inspiration is any one of the libraries which integrate `garde` or `validator` (or some other validation library) with some web framework's "extractors", e.g. `axum_garde`:

https://github.com/jprochazk/garde/blob/main/integrations/axum_garde/src/with_validation.rs#L95-L101
